src/frame/settings.rs

```diff
@@ -182,10 +182,10 @@ impl Settings {
                     }
                 }
                 Some(MaxFrameSize(val)) => {
-                    if val < DEFAULT_MAX_FRAME_SIZE || val > MAX_MAX_FRAME_SIZE {
-                        return Err(Error::InvalidSettingValue);
-                    } else {
+                    if (DEFAULT_MAX_FRAME_SIZE..=MAX_MAX_FRAME_SIZE).contains(&val) {
```

Comment:
Same here

Reply:
And same response.

Though it's probably worth keeping the flipped if. I had to apply de morgan's rule to understand what that really meant at first which is never a good sign.